torch/utils/pytree.py

```diff
@@ -27,8 +26,16 @@
     Union,
 )
 
-import optree
-from optree import PyTreeSpec  # direct import for type annotations
+from ._pytree import TreeSpec as PyTreeSpec
+
```

Comment:
This PR introduces a public torch.utils.pytree API?

Reply:
The previous PR introduced the public `torch.utils.pytree` API. But it is not imported in `torch/__init__.py`.

Reply:
It's OK. I will split this into smaller ones.